haskell-src/Concordium/Types/Parameters.hs

```diff
@@ -744,19 +1611,30 @@ instance HasExchangeRates (ChainParameters' cpv) where
 instance HasRewardParameters (ChainParameters' cpv) cpv where
     rewardParameters = cpRewardParameters
 
-putChainParameters :: IsChainParametersVersion cpv => Putter (ChainParameters' cpv)
+-- |Serialize a 'ChainParameters''.
+putChainParameters :: forall cpv. IsChainParametersVersion cpv => Putter (ChainParameters' cpv)
 putChainParameters ChainParameters{..} = do
-    put _cpElectionDifficulty
+    withIsConsensusParametersVersionFor (chainParametersVersion @cpv) $ put _cpConsensusParameters
     put _cpExchangeRates
     putCooldownParameters _cpCooldownParameters
-    putTimeParameters _cpTimeParameters
+    put _cpTimeParameters
     put _cpAccountCreationLimit
     put _cpRewardParameters
     put _cpFoundationAccount
     putPoolParameters _cpPoolParameters
 
+-- |Deserialize a 'ChainParameters''.
 getChainParameters :: forall cpv. IsChainParametersVersion cpv => Get (ChainParameters' cpv)
-getChainParameters = ChainParameters <$> get <*> get <*> getCooldownParameters <*> getTimeParameters <*> get <*> get <*> get <*> getPoolParameters
+getChainParameters =
+    ChainParameters
+        <$> withIsConsensusParametersVersionFor (chainParametersVersion @cpv) get
+        <*> get
+        <*> withIsCooldownParametersVersionFor (chainParametersVersion @cpv) get
+        <*> get
+        <*> get
+        <*> get
+        <*> get
+        <*> withIsPoolParametersVersionFor (chainParametersVersion @cpv) get
```

Comment:
Please don't do this. Name the variables you are parsing. This is always so hard to maintain since you have to count the order to figure out what is what.